portalnet/src/overlay/service.rs

```diff
@@ -2461,6 +2588,82 @@ fn pop_while_ssz_bytes_len_gt(enrs: &mut Vec<SszEnr>, max_size: usize) {
     }
 }
 
+/// References to `OverlayService` components required for processing
+/// a utp stream. This is basically a utility struct to avoid passing
+/// around a large number of individual references.
+struct UtpProcessing<TValidator, TStore, TContentKey>
+where
+    TContentKey: 'static + OverlayContentKey + Send + Sync,
+    TValidator: Validator<TContentKey>,
+    TStore: ContentStore,
+{
+    validator: Arc<TValidator>,
+    store: Arc<RwLock<TStore>>,
+    metrics: OverlayMetricsReporter,
+    kbuckets: Arc<RwLock<KBucketsTable<NodeId, Node>>>,
+    command_tx: UnboundedSender<OverlayCommand<TContentKey>>,
+    utp_controller: Arc<UtpController>,
+    accept_queue: Arc<RwLock<AcceptQueue<TContentKey>>>,
+}
+
+impl<TContentKey, TMetric, TValidator, TStore>
+    From<&OverlayService<TContentKey, TMetric, TValidator, TStore>>
+    for UtpProcessing<TValidator, TStore, TContentKey>
+where
+    TContentKey: 'static + OverlayContentKey + Send + Sync,
+    TValidator: Validator<TContentKey>,
+    TStore: ContentStore,
+{
+    fn from(service: &OverlayService<TContentKey, TMetric, TValidator, TStore>) -> Self {
+        Self {
+            validator: Arc::clone(&service.validator),
+            store: Arc::clone(&service.store),
+            metrics: service.metrics.clone(),
+            kbuckets: Arc::clone(&service.kbuckets),
+            command_tx: service.command_tx.clone(),
+            utp_controller: Arc::clone(&service.utp_controller),
+            accept_queue: Arc::clone(&service.accept_queue),
+        }
+    }
+}
+
+impl<TValidator, TStore, TContentKey> Clone for UtpProcessing<TValidator, TStore, TContentKey>
+where
+    TContentKey: 'static + OverlayContentKey + Send + Sync,
+    TValidator: Validator<TContentKey>,
+    TStore: ContentStore,
+{
+    fn clone(&self) -> Self {
+        Self {
+            validator: Arc::clone(&self.validator),
+            store: Arc::clone(&self.store),
+            metrics: self.metrics.clone(),
+            kbuckets: Arc::clone(&self.kbuckets),
+            command_tx: self.command_tx.clone(),
+            utp_controller: Arc::clone(&self.utp_controller),
+            accept_queue: Arc::clone(&self.accept_queue),
+        }
+    }
+}
```

Comment:
Should these be using `'static` for the reason I mentioned before, maybe arc's are a better option?